dwave/gate/simulator/simulator.pyx

```diff
@@ -41,7 +46,11 @@ def apply_instruction(
     little_endian: bool,
     conjugate_gate: bool = False,
 ):
-    if isinstance(op, ops.SWAP):
+    # if conditional and all bits are 1, skip
+    if op._cond and not any(op._cond):
+        return
```

Comment:
Easier to parse (so no comment needed):
```suggestion
    if op._cond and all(op._cond):
        return
```

Reply:
I'm realizing that both the code and the comment is wrong here. It should enter if at least one of the elements is `False`, so it should probably be `not all(op._cond)`. Thanks for highlighting that! 😅

Reply:
Without actually checking what `op._cond` is :see_no_evil:, if this is a common condition, perhaps add a helper method to `Operation`?

Reply:
Good point. I've added an `is_blocked` method [here](https://github.com/dwavesystems/dwave-gate/pull/12/files#diff-295c264653199148252d266a754a5553c61b034111d8504405adc2ad754ab60eR313-R322) instead.

Reply:
Perhaps make it a property (if you won't need future args)?